files/common/var/lib/delphix-platform/os-migration

```diff
@@ -63,10 +63,21 @@ function perform_migration() {
 		# In most cases, use the devices' by-id links to import
 		# domain0. This ensures persistent vdev names which can
 		# tolerate device reordering. AWS does not provide a by-id
-		# link, so we use simple device names (i.e. /dev/xvda)
+		# link, so we use simple device names (i.e. /dev/xvda). On
+		# Azure, however, we use the Azure-specific links, because the
+		# by-id links can change when the device is resized.
 		#
 		BY_ID_LINK="/dev/disk/by-id"
-		if [[ -d "$BY_ID_LINK" ]]; then
+		AZURE_LINK="/dev/disk/azure"
+		if [[ -d "$AZURE_LINK" ]]; then
+			dir_array=()
+			while read -r dir; do
+				dir_array+=(-d)
+				dir_array+=("$dir")
+			done < <(find "$AZURE_LINK" -maxdepth 1 -name "scsi*")
```

Comment:
I'm not sure azure supports more than 1 scsi controller. Have you seen instances where we have anything other than `scsi1`?

Reply:
I haven't, but given that it numbers the controller, I figured I would build the logic to handle it, since I wasn't at all sure either way. I can strip that out if you like; I'd also strip a little logic out of the app-gate review that handles it as well.

Reply:
I *do* know that the udev rules for azure have logic to create a scsi2 and scsi3 folder, but I don't know if those are ever triggered in practice.